misoc/integration/builder.py

```diff
@@ -169,7 +169,17 @@ def _initialize_rom(self):
                     w = boot_file.read(4)
                     if not w:
                         break
+                    if self.soc.cpu_type == "vexriscv":
+                        # Little endian swap
+                        w = w[::-1]
                     boot_data.append(struct.unpack(">I", w)[0])
+
+            if self.soc.cpu_type == "vexriscv":
+                # rom CRC endian reswap
+                buf = 0
+                for b in range(4):
+                    buf |= ((boot_data[-1] >> 8*b) & 0xFF) << (24-8*b)
+                boot_data[-1] = buf
```

Comment:
Can't this be done externally e.g. by objcopy?

Reply:
objcopy doesn't have anything for this, and in any case I don't see how bringing in *more* external dependencies makes anything better. This implementation is fine.

Reply:
We already use objcopy here. I'm surprised that it can only produce an image with the wrong endianness.

Reply:
Oh right; then I think it's better to change the CRC generation function to add an option to select endianness.

Reply:
@Dolu1990 Can you get to the bottom of the various endianness issues and propose minimal, clean code?

Reply:
BIOS CRC is added here:
https://github.com/m-labs/misoc/blob/master/misoc/software/bios/Makefile
https://github.com/m-labs/misoc/blob/master/misoc/tools/mkmscimg.py